### PR TITLE
api: swagger: use explicit definitions for some response types, and move examples per-field

### PR DESCRIPTION
**Pull request by thaJeztah**
This makes the definition of these types more explicit, and slightly easier to maintain. There's some follow-ups coming to this, which address CI flakiness when generating some response types (https://github.com/moby/moby/issues/42593), but those introduce some changes in the Go code, so keeping that separate for now.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Actually; added one more commit, which should help with https://github.com/moby/moby/issues/42593

**Comment by thaJeztah** [comment]
Arf.. I tried to split it, but this doesn't work, due to how the swagger generator works (expects the types to be in the same package), so I'll have to bring in more of the changes I had. Let me move the first two commits to a separate PR, as those changes are ok.

**Review by rumpl — APPROVED** [review]

File: api/types/volume/volume_list_o_k_body.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package volume
```
Comment (tianon) [inline api/types/volume/volume_list_o_k_body.go:1]: IMO this filename is a little funny (`api/types/volume/volume_list_o_k_body.go`) -- shouldn't `_o_k_` be just `_ok_`?

**Review by tianon — APPROVED** [review]

File: api/types/volume/volume_list_o_k_body.go (hunk 1 above)
Comment (thaJeztah) [reply api/types/volume/volume_list_o_k_body.go:1]: Yeah, it's horrible; I do have some follow ups to have more sensible names for some types; my intent with this PR was initially to try keep the types the same (but unfortunately had no way to influence the filename that was generated)

**Comment by thaJeztah** [comment]
Thx; bringing this one in 👍

File: api/types/volume/volume_list_o_k_body.go (hunk 1 above)
Comment (thaJeztah) [reply api/types/volume/volume_list_o_k_body.go:1]: -> https://github.com/moby/moby/pull/43530

**Pull request merged**